api/src/convert/signed_contingent_input.rs

```diff
@@ -0,0 +1,159 @@
+
+        let fpr = MockFogResolver::default();
+
+        // Charlie makes a signed contingent input, offering 1000 token2's for 1 MOB
+        let input_credentials = get_input_credentials(
+            BlockVersion::MAX,
+            Amount::new(1000, token2),
+            &charlie,
+            &fpr,
+            &mut rng,
+        );
+        let proofs = input_credentials.membership_proofs.clone();
+        let mut sci_builder = SignedContingentInputBuilder::new(
+            BlockVersion::MAX,
+            input_credentials,
+            fpr.clone(),
+            EmptyMemoBuilder::default(),
+        )
+        .unwrap();
+
+        // Originator requests an output worth 1MOB destined to themselves
+        sci_builder
+            .add_partial_fill_output(
+                Amount::new(1000 * MILLIMOB_TO_PICOMOB, Mob::ID),
+                &charlie.default_subaddress(),
+                &mut rng,
+            )
+            .unwrap();
+
+        // Change amount matches the input value
+        sci_builder
+            .add_partial_fill_change_output(
+                Amount::new(1000, token2),
+                &ReservedSubaddresses::from(&charlie),
+                &mut rng,
+            )
+            .unwrap();
+        let mut sci = sci_builder.build(&NoKeysRingSigner {}, &mut rng).unwrap();
+
+        sci.tx_in.proofs = proofs;
+        sci.tx_out_global_indices = vec![1, 2, 3, 4];
+
+        // decode(encode(sci)) should be the identity function.
+        {
+            let bytes = mc_util_serial::encode(&sci);
+            let recovered_sci = mc_util_serial::decode(&bytes).unwrap();
+            assert_eq!(sci, recovered_sci);
+        }
+
+        // Converting mc_transaction_extra::SignedContingentInput ->
+        // external::SignedContingentInput ->
+        // mc_transaction_extra::SignedContingentInput should be the identity
+        // function.
+        {
+            let external_sci: external::SignedContingentInput =
+                external::SignedContingentInput::from(&sci);
+            let recovered_sci: SignedContingentInput =
+                SignedContingentInput::try_from(&external_sci).unwrap();
+            assert_eq!(sci, recovered_sci);
+        }
+
+        // Encoding with prost, decoding with protobuf should be the identity function.
+        {
+            let bytes = mc_util_serial::encode(&sci);
+            let recovered_sci = external::SignedContingentInput::parse_from_bytes(&bytes).unwrap();
+            assert_eq!(recovered_sci, external::SignedContingentInput::from(&sci));
+        }
+
+        // Encoding with protobuf, decoding with prost should be the identity function.
+        {
+            let external_sci: external::SignedContingentInput =
+                external::SignedContingentInput::from(&sci);
+            let bytes = external_sci.write_to_bytes().unwrap();
+            let recovered_sci: SignedContingentInput = mc_util_serial::decode(&bytes).unwrap();
```

Comment:
:upside_down_face: I'm kind of surprised this one can be omitted, must be eliding type by the `assert_eq` below
```suggestion
            let recovered_sci = mc_util_serial::decode(&bytes).unwrap();
```